src/qmapshack/gis/CGisWorkspace.cpp

```diff
@@ -63,6 +64,18 @@ CGisWorkspace::CGisWorkspace(QMenu *menuProject, QWidget *parent)
 
     SETTINGS;
     treeWks->header()->restoreState(cfg.value("Workspace/treeWks/state", treeWks->header()->saveState()).toByteArray());
+
+    tags_hidden_e tagsHidden = (tags_hidden_e)cfg.value("Workspace/treeWks/tagsHidden", eTagsHiddenUnknown).toInt();
+    // Limit the column width to half the table width to avoid the name column to be pushed out the window.
+    if(tagsHidden == eTagsHiddenUnknown || treeWks->columnWidth(CGisListWks::eColumnRating) > 0.4 * this->width())
+    {
+        treeWks->setColumnWidth(CGisListWks::eColumnRating, 0.2 * this->width());
+    }
```

Comment:
Can you explain why you compare to 0.4 but limit to 0.2?

Reply:
I compare to 0.4, since that is approx the size at which the name gets pushed out. However it should default to a small value (0.2 seems good on my screen and since it is a ratio I hope it is on other screens too).

The difference in the two values should make sure the problem you described with the version before does not occur.